molgenis-data/src/main/java/org/molgenis/data/support/EntityDecorator.java

```diff
@@ -0,0 +1,340 @@
+package org.molgenis.data.support;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.StreamSupport.stream;
+
+import java.sql.Timestamp;
+import java.text.ParseException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.molgenis.MolgenisFieldTypes;
+import org.molgenis.MolgenisFieldTypes.FieldTypeEnum;
+import org.molgenis.data.AttributeMetaData;
+import org.molgenis.data.DataConverter;
+import org.molgenis.data.DataService;
+import org.molgenis.data.Entity;
+import org.molgenis.data.EntityMetaData;
+import org.molgenis.data.MolgenisDataException;
+import org.molgenis.data.UnknownAttributeException;
+import org.molgenis.fieldtypes.FieldType;
+import org.molgenis.fieldtypes.XrefField;
+import org.molgenis.util.MolgenisDateFormat;
+
+/**
+ * {@link Entity} decorator that uses an entity with different {@link EntityMetaData}. Values are converted on demand to
+ * the data types defined in the new EntityMetaData.
+ */
+public class EntityDecorator implements Entity
```

Comment:
Oops, originally this was intended to replace DefaultEntityImporter. But I got stuck in some weird self-reference code constructions. Class is removed.